tests/trainer/test_trainer_tricks.py

```diff
@@ -257,3 +258,19 @@ def test_error_on_dataloader_passed_to_fit(tmpdir):
 
     with pytest.raises(MisconfigurationException):
         trainer.fit(model, **fit_options)
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="test requires GPU machine")
+@pytest.mark.skipif(not NATIVE_AMP_AVALAIBLE, reason="test requires native AMP.")
+def test_auto_scale_batch_size_with_amp(tmpdir):
+    model = EvalModelTemplate()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_steps=1,
+        auto_scale_batch_size=True,
+        gpus=1,
+        precision=16
+    )
+    trainer.fit(model)
+    assert trainer.amp_backend == AMPType.NATIVE
+    assert trainer.scaler is not None
```

Comment:
Could we add an assert to check that the batch size have been altered?
Add `before_batch_size = model.batch_size` before training
Add `after_batch_size = model.batch_size` after training
Add `assert before_batch_size != after_batch_size` in the end

Reply:
yep thanks. see new commit

Reply:
Nice :+1: